app-foundation/4-projects/shared/app_cicd.tf

```diff
@@ -47,5 +70,26 @@ module "app_cicd_project" {
   business_code     = "bu1"
 }
 
+module "app_cicd_build_sa" {
+  source        = "terraform-google-modules/service-accounts/google"
+  version       = "~> 3.0"
+  project_id    = module.app_cicd_project.project_id
+  names         = ["cicd-build-sa"]
+  project_roles = local.sa_roles
+}
 
+resource "google_secret_manager_secret" "cicd_build_gsa_key_secret" {
+  project   = module.app_cicd_project.project_id
+  secret_id = "cicd_build_sa_key"
+  labels = {
+    label = "cicd-build-sa-key"
+  }
+  replication {
+    automatic = true
+  }
+}
 
+resource "google_secret_manager_secret_version" "cicd-build-gsa-key-secret-version" {
+  secret      = google_secret_manager_secret.cicd_build_gsa_key_secret.id
+  secret_data = "key_file" # Should be 'module.app_cicd_build_sa.key' if allowed by org-policy
```

Comment:
I am not sure why this is necessary?

Reply:
can take it out, Mike did it in his code, I will verify that the cloudbuild.yaml does not have a dependency and remove if none